### DISCUSSION TITLE
Registering SQL tables

### DISCUSSION THREAD

**Discussion by Siniara** [featureform/featureform#1126] [Q&A]
Hi, 

I'm going to preface this by saying that I'm a novice when it comes to feature stores and am in the process of learning about them, so the answer to this might be painfully obvious :)

My primary data source is a Teradata SQL data warehouse, which hosts databases that I use from which I retrieve certain tables. 
Can these tables be registered on featureform? I'm not sure I get it from the documentation. It says that any SQL table can be registered https://sdk.featureform.com/primary_sources/, but then I'm not sure how to do that, because the same page links to supported providers and Teradata isn't on the list, so that got me thinking that this isn't actually possible? 

Does a data source have to be supported to be usable with featureform? It would make sense I suppose, but one of the claims is that featureform is infrastructure agnostic, and then I'm not sure what would it mean. Or is there a process for adding other data sources not listed as the supported ones?

**Comment by mallman** [comment]
I would take a dive into the source code for the postgres/redshift/snowflake offline providers and see how they're implemented. It seems you can implement your own provider using Featureform's provider abstraction.
